src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp

```diff
@@ -87,7 +87,7 @@ class BiotPorosityUpdates : public PorosityBaseUpdates
                                                real64 & dPorosity_dTemperature ) const
   {
     real64 const biotSkeletonModulusInverse = (m_biotCoefficient[k] - m_referencePorosity[k]) / m_grainBulkModulus;
-    real64 const porosityThermalExpansion = 3 * m_thermalExpansionCoefficient[k] * m_biotCoefficient[k];
+    real64 const porosityThermalExpansion = 3 * m_thermalExpansionCoefficient[k] * ( m_biotCoefficient[k] - m_referencePorosity[k] );
```

Comment:
you sure about this whole thing?

Reply:
I am following the book by Coussy (2004):

<img width="735" alt="Screen Shot 2023-07-03 at 9 40 22 AM" src="https://github.com/GEOS-DEV/GEOS/assets/20196406/97bd414b-2f54-4f7b-a29b-0ed78d5b6225">

where the thermal expansion of the porosity is given by:

<img width="735" alt="Screen Shot 2023-07-03 at 9 40 57 AM" src="https://github.com/GEOS-DEV/GEOS/assets/20196406/b4b4bc74-7539-4b96-9e85-4eda4bb73162">

and in particular equations (4.21c) and (4.44)

Reply:
makes sense, thank you
@binw1 helped me to double check and it looks good

Reply:
@francoishamon We further reviewed the formula here:
porosity = porosity_n
               + biotCoefficient * meanStressIncrement / bulkModulus // change due to stress increment
               + biotSkeletonModulusInverse * deltaPressureFromBeginningOfTimeStep // change due to pressure increment
               - porosityThermalExpansion * deltaTemperatureFromBeginningOfTimeStep // change due to temperature increment
               + fixedStressPressureCoefficient * deltaPressureFromLastIteration // fixed-stress pressure term
               + fixedStressTemperatureCoefficient * deltaTemperatureFromLastIteration; // fixed-stress temperature term

And it seems the first term on the RHS should be biotCoefficient * (volumetricStrain_k - volumetricStrain_n), could you please double check it? Thanks.

Reply:
> @francoishamon We further reviewed the formula here: porosity = porosity_n + biotCoefficient * meanStressIncrement / bulkModulus // change due to stress increment + biotSkeletonModulusInverse * deltaPressureFromBeginningOfTimeStep // change due to pressure increment - porosityThermalExpansion * deltaTemperatureFromBeginningOfTimeStep // change due to temperature increment + fixedStressPressureCoefficient * deltaPressureFromLastIteration // fixed-stress pressure term + fixedStressTemperatureCoefficient * deltaTemperatureFromLastIteration; // fixed-stress temperature term
> 
> And it seems the first term on the RHS should be biotCoefficient * (volumetricStrain_k - volumetricStrain_n), could you please double check it? Thanks.

Hi @binw1, I think it should be `meanStressIncrement` here instead of the strain as we are using the fixed stress algorithm for the sequential solver. Note that this `computePorosity` function is only used for the sequential solver. 

@francoishamon maybe it is more precise if we change `meanStressIncrement` to `meanTotalStressIncrement` to indicate that we are using the fixed total stress algorithm.

Reply:
@binw1  and @frankfeifan regarding this comment and [the following comment](https://github.com/GEOS-DEV/GEOS/pull/2554#discussion_r1253474355): the two options are doing the same thing unless I am missing something, we just have to decide which one we prefer:

Option 1: pass the mean total stress increment
```
porosity = porosity_n 
+ biotCoefficient * meanTotalStressIncrement / bulkModulus // change due to stress increment 
+ biotSkeletonModulusInverse * (pressure - pressure_n) // change due to pressure increment 
- porosityThermalExpansion * (temperature - temperature_n) // change due to temperature increment 
+ fixedStressPressureCoefficient * (pressure - pressure_n) // fixed-stress pressure term 
+ fixedStressTemperatureCoefficient * (temperature - temperature_n); // fixed-stress temperature term
```
where
```
meanTotalStressIncrement = meanEffectiveStressIncrement 
- biotCoefficient * (pressure_k - pressure_n ) 
- 3 alpha ( temperature_k  - temperature_n ) 
```

Option 2: pass the mean effective stress increment
```
porosity = porosity_n 
+ biotCoefficient * meanEffectiveStressIncrement / bulkModulus // change due to stress increment 
+ biotSkeletonModulusInverse * (pressure - pressure_n) // change due to pressure increment 
- porosityThermalExpansion * (temperature - temperature_n) // change due to temperature increment 
+ fixedStressPressureCoefficient * (pressure_k - pressure_n) // fixed-stress pressure term 
+ fixedStressTemperatureCoefficient * (temperature_k - temperature_n); // fixed-stress temperature term
```
where
```
meanEffectiveStressIncrement = bulkModulus * ( strainIncrement[0] + strainIncrement[1] + strainIncrement[2] )
```

and I agree with @frankfeifan that we can be more precise on the naming by using the terms `effective` or `total` to qualify the stress.

My preference goes to Option 2 because it makes it easier to understand the time levels of pressure/temperature terms (rather than have pressure and temperature buried in the stress).

Reply:
Option 2 (implementation in this branch) is like that:
```
porosity = porosity_n
               + biotCoefficient * meanStressIncrement_k / bulkModulus // change due to stress increment (at the previous sequential
                                                                       // iteration)
               + biotSkeletonModulusInverse * (pressure - pressure_n) // change due to pressure increment
               - porosityThermalExpansion * (temperature - temperature_n) // change due to temperature increment
               + fixedStressPressureCoefficient * (pressure - pressure_k) // fixed-stress pressure term
               + fixedStressTemperatureCoefficient * (temperature - temperature_k); // fixed-stress temperature term
```
the last two terms are different from what @francoishamon wrote above. Both options are good, I just slightly prefer Option 2 for historical reasons.

Reply:
Yes, sounds good, it is what I implemented in the current version of this branch. I just fixed what is written in my previous message, sorry about the confusion :)

Reply:
> @binw1 and @frankfeifan regarding this comment and [the following comment](https://github.com/GEOS-DEV/GEOS/pull/2554#discussion_r1253474355): the two options are doing the same thing unless I am missing something, we just have to decide which one we prefer:
> 
> Option 1: pass the mean total stress increment
> 
> ```
> porosity = porosity_n 
> + biotCoefficient * meanTotalStressIncrement / bulkModulus // change due to stress increment 
> + biotSkeletonModulusInverse * (pressure - pressure_n) // change due to pressure increment 
> - porosityThermalExpansion * (temperature - temperature_n) // change due to temperature increment 
> + fixedStressPressureCoefficient * (pressure - pressure_n) // fixed-stress pressure term 
> + fixedStressTemperatureCoefficient * (temperature - temperature_n); // fixed-stress temperature term
> ```
> 
> where
> 
> ```
> meanTotalStressIncrement = meanEffectiveStressIncrement 
> - biotCoefficient * (pressure_k - pressure_n ) 
> - 3 alpha ( temperature_k  - temperature_n ) 
> ```
> 
> Option 2: pass the mean effective stress increment
> 
> ```
> porosity = porosity_n 
> + biotCoefficient * meanEffectiveStressIncrement / bulkModulus // change due to stress increment 
> + biotSkeletonModulusInverse * (pressure - pressure_n) // change due to pressure increment 
> - porosityThermalExpansion * (temperature - temperature_n) // change due to temperature increment 
> + fixedStressPressureCoefficient * (pressure - pressure_k) // fixed-stress pressure term 
> + fixedStressTemperatureCoefficient * (temperature - temperature_k); // fixed-stress temperature term
> ```
> 
> where
> 
> ```
> meanEffectiveStressIncrement = bulkModulus * ( strainIncrement[0] + strainIncrement[1] + strainIncrement[2] )
> ```
> 
> and I agree with @frankfeifan that we can be more precise on the naming by using the terms `effective` or `total` to qualify the stress.
> 
> My preference goes to Option 2 because it makes it easier to understand the time levels of pressure/temperature terms (rather than have pressure and temperature buried in the stress).

Yeah. I also agree that Option 2 is clearer and better.